src/polhemus_tf_broadcaster.cpp

```diff
@@ -0,0 +1,292 @@
+
+  // Retrieve launch parameters
+  nstations_attached = nstations_broadcast = 0;
+  for (i = 0; i < 8; i++) {
+    bool station_attached, station_broadcast;
+    s = i + 1;
+
+    nh.getParam("/station_"+std::to_string(s)+"_attached", station_attached);
+    nh.getParam("/station_"+std::to_string(s)+"_broadcast", station_broadcast);
+
+    broadcast_stations[i] = station_broadcast;
+
+    if (station_attached) {
+      nstations_attached += 1;
+      fprintf(stderr, "  Station %d attached.\n", s);
+    }
+
+    if (station_broadcast) {
+      nstations_broadcast += 1;
+      fprintf(stderr, "  Broadcasting station %d.\n\n", s);
+    }
+    else if (station_attached) {
+      fprintf(stderr, "\n");
+    } 
+  }
+
+  if (nstations_found != nstations_attached) {
+    fprintf(stderr, "Number of stations found (%d) does not match the number of stations attached (%d).\n", nstations_found, nstations_attached);
+    usb_close(handle);
+    return 1;
+  }
+
+  /* define which information to get per sensor (called a station
+     by polhemus)
+
+     o* applies to all stations
+
+     if this is changed, the euler_station_t struct or (***) below has to be edited accordingly 
+  */
+  liberty_send(handle, (char *)"O*,8,9,11,3,5\r"); // euler_station_t: euler angles
+
+  /* set output hemisphere -- this will produce a response which we're
+     ignoring 
+  */
+  nh.getParam("/x_hs", x_hs);
+  nh.getParam("/y_hs", y_hs);
+  nh.getParam("/z_hs", z_hs);
+  set_hemisphere(handle, x_hs, y_hs, z_hs);
+
+  /* switch output to centimeters */
+  //liberty_send(handle, "u1\r");
+  liberty_clear_input(handle); //right now, we just ignore the answer
+
+  // (***)
+  euler_station_t *stations = (euler_station_t*) (malloc(sizeof(euler_station_t) * nstations_found));
+  if (!stations)
+    abort();
+
+  /* set up signal handler to catch the interrupt signal */
+  signal(SIGINT, signal_handler);
+
+  go_on = 1;
+
+  /* enable continuous mode (get data points continously) */
+  liberty_send(handle, (char *)"c\r");
+
+  gettimeofday(&tv, NULL);
+  printf("Begin time: %d.%06d\n", (unsigned int) (tv.tv_sec), (unsigned int) (tv.tv_usec));
+  
+  // Start main loop
+  while(ros::ok()) {
+
+    // Update polhemus
+    // (***)
+    if (!liberty_receive(handle, &buf, stations, sizeof(euler_station_t) * nstations_found)) {
+      fprintf(stderr, "Recieve failed.\n");
+      return 2;
+    }
+
+    /* Note: timestamp is the time in mS after the first read to the
+       system after turning it on
+       at 240Hz, the time between data sample is 1/240 = .00416_
+       seconds.
+       The framecount is a more exact way of finding out the time:
+       timestamp = framecount*1000/240 (rounded down to an int)* 
+    */
+
+    for (i=0; i < 8; i++) {
+      if (broadcast_stations[i]) {
+	s = i + 1;
+	static tf::TransformBroadcaster br;
+	tf::Transform transform;
+
+	// Set translation
+	transform.setOrigin(tf::Vector3(0.0254*stations[i].x,
```

Comment:
This is converting inches to cm I presume? Please add a comment explaining "magic" conversions.